### PR TITLE
akka-cluster-sharding compiler warnings as fatal errors

### PR DESCRIPTION
**Pull request by johanandren**
Refs #26088

### CODE REVIEW

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala [hunk 1]
```diff
@@ -129,22 +130,14 @@ private[akka] object Shard {
...
-      Props(
-        new PersistentShard(
-          typeName,
-          shardId,
-          entityProps,
-          settings,
-          extractEntityId,
-          extractShardId,
-          handOffStopMessage)).withDeploy(Deploy.local)
+      Props(new PersistentShard(typeName, shardId, entityProps, settings, extractEntityId, handOffStopMessage))
```
Comment (raboof) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala]: So the scalafmt sbt plugin leaves this be, but the autoformatter puts it all on one line anyway? Weird.

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala [hunk 2]
```diff
@@ -460,7 +452,7 @@ private[akka] class Shard(
     }
   }
 
-  def deliverTo(id: EntityId, msg: Any, payload: Msg, snd: ActorRef): Unit = {
+  def deliverTo(id: EntityId, @unused msg: Any, payload: Msg, snd: ActorRef): Unit = {
```
Comment (raboof) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala]: Since this is internal API, can we just remove that parameter entirely?

**Review by raboof — APPROVED** [review]
Nice!

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala (hunk 2 above)
Comment (johanandren) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala]: It is used in an overload, so no.

**Review by helena — APPROVED** [review]
LGTM!

**Comment by johanandren** [comment]
I think it is only the scala/javadoc links failing now.

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala [hunk 3]
```diff
@@ -749,15 +748,16 @@ private[akka] class ClusterShardingGuardian extends Actor {
...
-                childProps = coordinatorProps,
-                childName = "coordinator",
-                minBackoff = coordinatorFailureBackoff,
-                maxBackoff = coordinatorFailureBackoff * 5,
-                randomFactor = 0.2,
-                maxNrOfRetries = -1)
-              .withDeploy(Deploy.local)
+            val singletonProps =
+              BackoffOpts
+                .onFailure(
```
Comment (patriknw) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala]: This was a change in behavior, which caused https://github.com/akka/akka/issues/26724 to fail. I think the old behavior corresponds to `onStop` (which may happen for persist failures). I'm fixing.

Comment (johanandren) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala]: dang, sorry!

**Pull request merged**